oci_spec/src/lib.rs

```diff
@@ -22,265 +37,491 @@ pub struct Box {
+        source: PathBuf::from("tmpfs"),
+        options: vec![
+            "nosuid".to_string(),
+            "strictatime".to_string(),
+            "mode=755".to_string(),
+            "size=65536k".to_string(),
+        ],
+    });
+
+    default_mounts.push(Mount {
+        destination: PathBuf::from("/dev/pts"),
+        typ: String::from("devpts"),
+        source: PathBuf::from("devpts"),
+        options: vec![
+            "nosuid".to_string(),
+            "noexec".to_string(),
+            "newinstance".to_string(),
+            "ptmxmode=0666".to_string(),
+            "mode=0620".to_string(),
+            "gid=5".to_string(),
+        ],
+    });
+
+    default_mounts.push(Mount {
+        destination: PathBuf::from("/dev/shm"),
+        typ: String::from("tmpfs"),
+        source: PathBuf::from("shm"),
+        options: vec![
+            "nosuid".to_string(),
+            "noexec".to_string(),
+            "nodev".to_string(),
+            "mode=1777".to_string(),
+            "size=65536k".to_string(),
+        ],
+    });
+
+    default_mounts.push(Mount {
+        destination: PathBuf::from("/dev/mqueue"),
+        typ: String::from("mqueue"),
+        source: PathBuf::from("mqueue"),
+        options: vec![
+            "nosuid".to_string(),
+            "noexec".to_string(),
+            "nodev".to_string(),
+        ],
+    });
+
+    default_mounts.push(Mount {
+        destination: PathBuf::from("/sys"),
+        typ: String::from("sysfs"),
+        source: PathBuf::from("sysfs"),
+        options: vec![
+            "nosuid".to_string(),
+            "noexec".to_string(),
+            "nodev".to_string(),
+            "ro".to_string(),
+        ],
+    });
+
+    default_mounts.push(Mount {
+        destination: PathBuf::from("/sys/fs/cgroup"),
+        typ: String::from("cgroup"),
+        source: PathBuf::from("cgroup"),
+        options: vec![
+            "nosuid".to_string(),
+            "noexec".to_string(),
+            "nodev".to_string(),
+            "relatime".to_string(),
+            "ro".to_string(),
+        ],
+    });
+
+    default_mounts
+}
+
+// Specifies UID/GID mappings
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct LinuxIdMapping {
+    // Starting uid/gid on the host to be mapped to container_id
     #[serde(default, rename = "hostID")]
     pub host_id: u32,
+    // Starting uid/gid in the container
     #[serde(default, rename = "containerID")]
     pub container_id: u32,
+    // Number of IDs to be mapped
     #[serde(default)]
     pub size: u32,
 }
 
-// a is for LinuxDeviceCgroup
-#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
+// Device types
+#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq)]
 #[serde(rename_all = "lowercase")]
 pub enum LinuxDeviceType {
+    // block (buffered)
     B,
+    // character (unbuffered)
     C,
+    // character (unbufferd)
     U,
+    // FIFO
     P,
+    // ??
```

Comment:
Do you know what A is for? and why does it default to A (It already had a default impl)